src/pylint_odoo/checkers/function_checker.py

```diff
@@ -0,0 +1,45 @@
+from astroid import ClassDef, FunctionDef
+from pylint.checkers import utils
+
+from .odoo_base_checker import OdooBaseChecker
+
+ODOO_MSGS = {
+    # C->convention R->refactor W->warning E->error F->fatal
+    "E8501": (
+        "No exceptions should be raised inside unlink() functions",
+        "no-raise-unlink",
+        "Use @api.ondelete to add any constraints instead",
+    ),
+}
+
+
+class FunctionChecker(OdooBaseChecker):
+    name = "odoolint"
+    msgs = ODOO_MSGS
+    checks_maxmin_odoo_version = {"no-raise-unlink": {"odoo_minversion": "15.0"}}
```

Comment:
Could you split this line, please? So it will be easier to add future lints or more conditions.

```suggestion
    checks_maxmin_odoo_version = {
        "no-raise-unlink": {
            "odoo_minversion": "15.0",
        },
    }
```

Besides, it's more readable.